src/CssInlinerFactory.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier;
+
+use DOMDocument;
+
+/**
+ * This class provides a factory for CssInliner that can be passed as a dependency
+ * or be injected via dependency injection
+ *
+ * @author SpazzMarticus <SpazzMarticus@users.noreply.github.com>
+ */
+class CssInlinerFactory
+{
+    /**
+     * Creates a new instance of CssInliner from a string
+     *
+     * @param string $unprocessedHtml
+     *
+     * @return CssInliner
+     *
+     * @throws \InvalidArgumentException if $unprocessedHtml is anything other than a non-empty string
+     */
+    public function createFromHtml(string $unprocessedHtml): CssInliner
```

Comment:
This now is limited to only `CssInliner`. Should we make this more flexible?

Reply:
I see multiple options here:

- Factory per class `CssInlinerFactory`, `HtmlPrunerFactory`, ...
- Factory with a method per class `createCssInlinerFromHtml($html)`, `createHtmlPrunerFromHtml($html)`
- Dynamic factory à la `createFromHtml($class,$html)`

I am not in favor of the third option, since checking if `$class` is a child of `AbstractHtmlProcessor` is necessary and returning a `AbstractHtmlProcessor` breaks typehinting to a specific class.